packages/graphql-language-service-server/src/languages.ts

```diff
@@ -0,0 +1,66 @@
+import { Position, Range } from 'graphql-language-service-utils';
+import { makeExtractTagsFromSource, GraphQLSource } from './languageUtils';
+
+export type LanguageConfig = {
+  readonly extensions: ReadonlyArray<string>;
+  readonly fileFilter: (content: string) => boolean;
+  readonly extractSources: (content: string) => ReadonlyArray<GraphQLSource>;
+};
+
+/**
+ * Filters and other regexps used later on.
+ */
+const jsFileFilter = new RegExp(/graphql`|graphql\.experimental`|gql`/);
+const jsGraphQLTagsRegexp = new RegExp(
+  /(?<=(graphql|gql|graphql\.experimental)`)[\s\S]+?(?=`)/g,
+);
+
+// This removes interpolation in JS strings, like ` ${someInterpolationHere}`
+const jsRemoveInterpolationRegexp = new RegExp(/([\s]+\${)[\s\S]+?(})/g);
+
+const jsExtractor = makeExtractTagsFromSource(jsGraphQLTagsRegexp);
+
+const reasonFileFilterRegexp = new RegExp(/(\[%(graphql|relay\.))/);
+const reasonGraphQLTagsRegexp = new RegExp(
+  /(?<=\[%(graphql|relay\.\w*)[\s\S]*{\|)[.\s\S]+?(?=\|})/g,
+);
+const reasonExtractor = makeExtractTagsFromSource(reasonGraphQLTagsRegexp);
+
+const jsLanguages: LanguageConfig = {
+  extensions: ['js', 'jsx', 'ts', 'tsx'],
```

Comment:
does this work for `.esm.js`, etc? also `.mjs` we need as well i think?

Reply:
Adding them to the tests!